soc/openhwgroup/cva6/soc_cache_management.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * Non-standard CVA6 cache management operations.
+ *
+ * Copyright(c) 2024, CISPA Helmholtz Center for Information Security
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stddef.h>
+#include <zephyr/arch/cache.h>
+#include <zephyr/arch/riscv/csr.h>
+#include <zephyr/kernel.h>
+
+#include "cva6.h"
+
+void arch_dcache_enable(void)
+{
+	csr_write(SOC_CVA6_CUSTOM_CSR_DCACHE, SOC_CVA6_CUSTOM_CSR_DCACHE_ENABLE);
+}
+
+void arch_dcache_disable(void)
+{
+	csr_write(SOC_CVA6_CUSTOM_CSR_DCACHE, SOC_CVA6_CUSTOM_CSR_DCACHE_DISABLE);
+}
+
+void arch_icache_enable(void)
+{
+	csr_write(SOC_CVA6_CUSTOM_CSR_ICACHE, SOC_CVA6_CUSTOM_CSR_ICACHE_ENABLE);
+}
+
+void arch_icache_disable(void)
+{
+	csr_write(SOC_CVA6_CUSTOM_CSR_ICACHE, SOC_CVA6_CUSTOM_CSR_ICACHE_DISABLE);
+}
```

Comment:
why is arch code being added to soc?

Reply:
This is non-standard and SOC-specific. RISC-V has an extension for cache management operations, but AFAIK it has not been ratified yet and is not implemented in CVA6. I will make them `__weak` such that it can be overwritten in the future.